File: math/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if  x < 0:
            return False

        mask = 1
        while x >= 10 * mask:
            mask *= 10
        
        while x > 0:
            lsd = x % 10
            msd = x // mask
            if lsd != msd:
                return False
            x %= mask
            x //= 10
            mask //= 100
        return True

File: math/palindrome-number.rb
# @param {Integer} x
# @return {Boolean}
def is_palindrome(x)
  return false if x.negative?

  mask = 1
  while x > (10 * mask)
    mask *= 10
  end

  while x > 0
    lsd = x % 10
    msd = x / mask
    return false if lsd != msd
    x = x % mask
    x = x / 10
    mask /= 100
  end
  true
end